war/src/main/webapp/scripts/hudson-behavior.js

```diff
@@ -2342,18 +2335,9 @@ function updateBuildHistory(ajaxUrl,nBuild) {
                 }
             });
         }
-
-        pageSearchInput.observe('keypress', function(e) {
-            var key = e.which || e.keyCode;
-            // On enter
-            if (key === 13) {
-                loadPage({}, true);
-            }
-        });
-        pageSearchClear.observe('click', function() {
-            pageSearchInput.value = '';
+        pageSearchInput.oninput = function() {
```

Comment:
I'd suggest to revert this. This would cause a lot of requests to the backend, and there's no sane way of adding a debounce to this file. Other option would be to extract all this functionality to a webpack entrypoint and use `lodash.debounce`

Reply:
I've extracted the build history JS from `hudson-behavior.js` and added debounce - the file itself is a little rough around the edges at the moment but its working well from the look of it.

Reply:
Ok, sounds good